### PR TITLE
is_default_branch global expression

### LINKED ISSUE #184: The suggested default branch example does not work for schedule
```yaml
tags: |
  type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
```

Here is an example:

https://github.com/felipecrs/jenkins-agent-dind/blob/master/.github/workflows/ci.yml#L35

https://github.com/felipecrs/jenkins-agent-dind/runs/5964649319?check_suite_focus=true#step:3:31

I would say that this variable does not get filled properly for schedule events, which is meaningful, as metadata-action relies on it.

### PR DESCRIPTION
**Pull request by crazy-max**
fixes #184

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -605,26 +611,28 @@ tags: |
...
-is not `master`, use `type=raw` with a boolean expression:
+For conditionally tagging with latest for a specific branch name, e.g. if your
+default branch name is not `master`, use `type=raw` with a boolean expression:
 
 ```yaml
 tags: |
-  type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+  # set latest tag for master branch
+  type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'master') }}
 ```
```
Comment (felipecrs) [inline README.md:621]: IMO you should remove this entirely. A solution which may not work under some circumstances should not be suggested, especially without warnings.

File: README.md [hunk 2]
```diff
@@ -634,6 +642,63 @@ tags: |
...
+
+Returns the base ref or target branch of the pull request that triggered the
+workflow run. Will be empty for a branch reference:
+
+| Event           | Ref                           | Output             |
+|-----------------|-------------------------------|--------------------|
+| `pull_request`  | `refs/pull/2/merge`           | `master`           |
+| `push`          | `refs/heads/master`           |                    |
+| `push`          | `refs/heads/my/branch`        |                    |
+| `push tag`      | `refs/tags/v1.2.3`            | `master`           |
```
Comment (felipecrs) [inline README.md:683]: I believe you should not populate this variable in tag events, because it will always be the default branch.

Tags aren't anyhow tied to a branch in Git.


Example:

```console
$ git checkout -b feature
$ git commit -m "Test commit" --allow-empty
$ git tag v1.2.3
$ git push --follow-tags
```

And the result is:

https://github.com/felipecrs/ghaction-dump-context/runs/6104549396?check_suite_focus=true#step:3:356

```console
$ git checkout master
$ git tag v1.2.3-master
$ git push origin v1.2.3-master
```

Results in:
https://github.com/felipecrs/ghaction-dump-context/runs/6104557753?check_suite_focus=true#step:3:356

**Review by felipecrs — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (crazy-max) [reply README.md:621]: I actually updated this example with a hardcoded branch name to show how to handle this case for a specific branch as it seems still valid for me.

File: README.md (hunk 2 above)
Comment (crazy-max) [reply README.md:683]: Agree that a tag points to a commit, and commits exist independently of branches. But GitHub context seems to record the branch where this event occurs: https://github.com/docker/metadata-action/blob/0950af0d7cc44517c4cf423a732b5ad02d839439/__tests__/fixtures/event_tag_v1.1.1.json#L3.

You can see it in your logs: https://github.com/felipecrs/ghaction-dump-context/runs/6104549396?check_suite_focus=true#step:3:360

File: README.md (hunk 1 above)
Comment (felipecrs) [reply README.md:621]: Oh, right. I didn't notice.

File: README.md (hunk 2 above)
Comment (felipecrs) [reply README.md:683]: Very interesting. Thank you.

**Review by felipecrs — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (felipecrs) [reply README.md:683]: BTW, I made both `master` and `feature` point to the same revision, and it does not matter how I tag (or push) them, GitHub seems to just select one randomly to be the `base_ref` in case both matches the same revision.

https://github.com/felipecrs/ghaction-dump-context/runs/6164266775?check_suite_focus=true#step:3:360

This was created with:

```console
$ git checkout master

$ git tag v1.2.3-sentfrommaster

$ git push origin v1.2.3-sentfrommaster
```

Comment (crazy-max) [reply README.md:683]: I would say it's GitHub' side bug:

* `v1.2.3-sentfromfeature` looks good: https://github.com/felipecrs/ghaction-dump-context/actions/runs/2222486218 and tagged against `32b85ec283cee8ce1a03419b190eab50a1be874d` revision.
* You then create another tag `v1.2.3-sentfrommaster` against the very same commit: https://github.com/felipecrs/ghaction-dump-context/actions/runs/2222492052

I suppose GitHub manages a DB that records the base ref of a tag. In this case you have created both tag against the same commit so I presume GitHub logic returns all tags for the same commit and matches the first one he found in the DB and therefore here the wrong ref is returned. Just speculations ofc but that seems to be it.

Maybe they `SELECT 1` and `ORDER by name` or smth like that. Can you try to tag from master for a tag named smth like `v1.2.3-aaaaa` against the same commit and see what it returns?
Can you also tag against another commit from master?

Comment (felipecrs) [reply README.md:683]: No luck with the first test:

https://user-images.githubusercontent.com/29582865/165166532-76729e38-7d61-4934-8378-3eafd0817a58.mp4

But the second brought something even more interesting:


https://user-images.githubusercontent.com/29582865/165166623-247b6c2c-be20-46c6-a16b-3c4e556d66fb.mp4

Comment (felipecrs) [reply README.md:683]: > I would say it's GitHub' side bug

I agree this isn't your problem. If GitHub decides to publish that information, then it's their responsibility to make it reliable or not. But have they documented that this can be used to reliably check the source branch of a tag?

Honestly speaking, if they did, they probably made a mistake, as I don't think this is actually possible.

Comment (crazy-max) [reply README.md:683]: Oh boy that's bad :sweat_smile:

Comment (crazy-max) [reply README.md:683]: So yeah for the second one it returns because a tag exists for the very same commit: https://github.com/felipecrs/ghaction-dump-context/releases/tag/v1.2.3-master

Comment (crazy-max) [reply README.md:683]: > > I would say it's GitHub' side bug
> 
> I agree this isn't your problem. If GitHub decides to publish that information, then it's their responsibility to make it reliable. But have they documented that this can be used to reliably check the source branch of a tag?
> 
> Honestly speaking, if they did, they probably made a mistake, as I don't think this is actually possible.

`base_ref` is displayed in their docs but not documented :thinking:: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push

Comment (crazy-max) [reply README.md:683]: Can you try to tag from master against a commit not linked to any tag?

Comment (felipecrs) [reply README.md:683]: I would suspect that the intended behavior for this attribute was something more internal to GitHub itself. Perhaps to calculate some cache validity of source code internally?

Comment (crazy-max) [reply README.md:683]: You triggered on a commit quite old that at that time didn't have this workflow availalbe I presume:

image

Comment (felipecrs) [reply README.md:683]: Wow, that would make sense. Let me try again.

Comment (crazy-max) [reply README.md:683]: Yep no tag event in your workflow for this commit: https://github.com/felipecrs/ghaction-dump-context/blob/4cf159ecaf43bd0f2487e3fb8f5fc723fce1d32c/.github/workflows/ci.yml

Comment (felipecrs) [reply README.md:683]: Ok, check my new tags. I believe the `base_ref` gets populated with the first reference in alphabetical order.

- `refs/heads/<branch>` will only be there if tags matches the latest revision of the branch
- `refs/heads/<branch>` will always get precedence over `refs/tags/<oldtag>` since it's alphabetically ordered
- That also explains why `refs/heads/feature` is always chosen instead of `master`, no matter what.

Comment (felipecrs) [reply README.md:683]: BTW, in case you publish a tag against a revision which does not point to the latest version of a tag neither another tag, here is what you get (null):

https://github.com/felipecrs/ghaction-dump-context/runs/6164973057?check_suite_focus=true#step:3:360

Comment (crazy-max) [reply README.md:683]: > BTW, in case you publish a tag against a revision which does not point to the latest version of a tag neither another tag, here is what you get (`null`):

Yes just saw that too and I don't like that. I will remove support of `base_ref` as it seems to be not what we want.

File: README.md [hunk 3]
```diff
@@ -634,6 +642,63 @@ tags: |
...
+
+#### `{{is_default_branch}}`
+
+Returns `true` if the branch that triggered the workflow run is the default
+one, otherwise `false`.
```
Comment (akostadinov) [inline README.md:688]: @crazy-max , could you clarify here, whether a pull-request to default branch will return true or only a `push` event to default branch will be `true`?
